examples/step-69/step-69.cc

```diff
@@ -2464,11 +2469,28 @@ namespace Step69
     pcout << "done" << std::endl;
 
     // Next we create the triangulation, assemble all matrices, set up
-    // scratch space, and initialize the DataOut<dim> object:
+    // scratch space, and initialize the DataOut<dim> object. All of these
+    // operations are pretty standard and discussed in detail in the
+    // Discretization and OfflineData classes.
+    //
+    // We have to make take care of a special case when resuming an
+    // interrupted computation though: In order to be able to read in the
+    // saved mesh and associated state vector we have to make sure to
+    // not refine the coarse mesh:
 
     {
       print_head(pcout, "create triangulation");
-      discretization.setup();
+
+      if (resume)
+        {
+          discretization.refinement = 0;
+          discretization.setup();
+          discretization.triangulation.load(base_name + "-checkpoint.mesh");
+        }
+      else
+        {
+          discretization.setup();
+        }
```

Comment:
```suggestion
      if (resume)
        {
          discretization.setup();
          discretization.triangulation.load(base_name + "-checkpoint.mesh");
        }
      else
        {
          discretization.setup();
          discretization.triangulation.refine_global(discretization.refinement);
        }
```

I would suggest to move the `refine_global()` part out of the `setup()` function. This way, it is more clear what `setup()` actually does in my opinion.

Technically both meshes are still refined `discretization.refinement` times. With the suggestion, there is no need to set it to 0.

Reply:
Sounds good, will do.